OSDOCS-10989: Documented the 4.15.19 z-stream release notes (#77953)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

// 4.15.19
[id="ocp-4-15-19"]
=== RHSA-2024:4041 - {product-title} 4.15.19 bug fix and security update

Issued: 2024-06-27

{product-title} release 4.15.19 is now available. The list of bug fixes that are included in this update is documented in the link:https://access.redhat.com/errata/RHSA-2024:4041[RHSA-2024:4041] advisory. The RPM packages that are included in this update are provided by the link:https://access.redhat.com/errata/RHBA-2024:4044[RHBA-2024:4044] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.19 --pullspecs
----

[id="ocp-4-15-19-bug-fixes"]
==== Bug fixes

> * Previously, when a new version of a Custom Resource Definition (CRD) specified a new conversion strategy, this conversion strategy was expected to successfully convert resources. This was not the case because Operator Lifecycle Manager (OLM) cannot run the new conversion strategies for CRD validation without actually performing the update operation. With this release, the OLM implementation generates a warning message during the update process when CRD validations fail with the existing conversion strategy and the new conversion strategy is specified in the new version of the CRD. (link:https://issues.redhat.com/browse/OCPBUGS-35720[*OCPBUGS-35720*]).

Reviewer: ```suggestion
* Previously, when a new version of a custom resource definition (CRD) specified a new conversion strategy, this conversion strategy was expected to successfully convert resources. This was not the case because Operator Lifecycle Manager (OLM) cannot run the new conversion strategies for CRD validation without actually performing the update operation. With this release, the OLM implementation generates a warning message during the update process when CRD validations fail with the existing conversion strategy and the new conversion strategy is specified in the new version of the CRD. (link:https://issues.redhat.com/browse/OCPBUGS-35720[*OCPBUGS-35720*]).
```
Nit that isn't worth blocking a merge on. See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#custom-resource-definition-crd

Author: Thanks. This can be addressed in the next iteration of 4.15.z.

> * Previously, when a new version of a custom resource definitions (CRD) specifies a conversion strategy, the assumption was that the new conversion strategy would successfully convert resources. This was not the case because Operator Lifecycle Manager (OLM) cannot run the new conversion strategies for Custom Resource Definition (CRD) validation without actually performing the upgrade operation. OLM cannot guarantee that an existing conversion strategy converts the same as the specified strategy. With this release, the OLM implementation generates a warning message during the upgrade process when CRD validations fail with the existing conversion strategy and the new conversion strategy is specified in the new version of the CRD. (link:https://issues.redhat.com/browse/OCPBUGS-35720[*OCPBUGS-35720*]).

Author: :heavy_check_mark:

> * Previously, during node reboots, especially during upgrade operations, the node that interacts with the rebooting machine enters a `Ready=Unknown` state for a short amount of time. This situation caused the Control Plane Machine Set Operator to enter an `UnavailableReplicas` condition and then an `Available=false` state. The `Available=false` state triggers alerts that demand urgent action, but intervention would only be required for a short period until the node reboots. With this release, a grace period for node unreadiness is provided where if a node enters an unready state, the Control Plane Machine Set Operator will not instantly enter an `UnavailableReplicas` condition or an `Available=false` state. (link:https://issues.redhat.com/browse/OCPBUGS-34971[*OCPBUGS-34971*]).

Author: :heavy_check_mark:

* Previously, the OpenShift Cluster Manager container did not have the right TLS Certificates. As a result, image streams could not be used in disconnected deployments. With this update, the TLS Certificates are added as projected volumes. (link:https://issues.redhat.com/browse/OCPBUGS-34580[*OCPBUGS-34580*])

> * Previously, when a serverless function was created in the create serverless form, BuilldConfig was not created. With this update, if the Pipelines Operator is not installed, if the pipeline resource is not created for particular resource, or if the pipeline is not added while creating a serverless function, it will create BuildConfig as expected. (link:https://issues.redhat.com/browse/OCPBUGS-34350[*OCPBUGS-34350*])

Reviewer 2: ```suggestion
* Previously, when a serverless function was created in the create serverless form, `BuildConfig` was not created. With this update, if the Pipelines Operator is not installed, or if the pipeline resource is not created for particular resource, or if the pipeline is not added while creating a serverless function, the `BuildConfig` is created as expected. (link:https://issues.redhat.com/browse/OCPBUGS-34350[*OCPBUGS-34350*])
```

Author: :heavy_check_mark:

> * Previously, the reduction of the network queue did not work as expected for inverted rules such as `!ens0`. This happened because the exclamation mark symbol was duplicated in the generated tuned profile. With this release, the duplication no longer occurs so that inverted rules would as intended. (link:https://issues.redhat.com/browse/OCPBUGS-33929[*OCPBUGS-33929*]).

Reviewer 2: ```suggestion
* Previously, the reduction of the network queue did not work as expected for inverted rules such as `!ens0`. This happened because the exclamation mark symbol was duplicated in the generated tuned profile. With this release, the duplication no longer occurs so that inverted rules apply as intended. (link:https://issues.redhat.com/browse/OCPBUGS-33929[*OCPBUGS-33929*]).
```

Author: :heavy_check_mark:

> * Previously, registry overrides applied to non-relevant data plane components. With this release, registry overrides no longer apply to these components. (link:https://issues.redhat.com/browse/OCPBUGS-33627[*OCPBUGS-33627*]).

Reviewer 2: ```suggestion
* Previously, registry overrides applied to non-relevant data-plane components. With this release, registry overrides no longer apply to these components. (link:https://issues.redhat.com/browse/OCPBUGS-33627[*OCPBUGS-33627*]).
```

Author: :heavy_check_mark:

> * Previously, when installing a cluster on {vmw-first}, the installation would fail if an ESXi host was in maintenance mode due to the installation program failing to retrieve version information from the host. With this update, the installation program does not attempt to retrieve version information from ESXi hosts that are in maintenance mode, allowing the installation to proceed. (link:https://issues.redhat.com/browse/OCPBUGS-31387[*OCPBUGS-31387*])

Reviewer 2: ```suggestion
* Previously, when installing a cluster on {vmw-first}, the installation failed if an ESXi host was in maintenance mode because the installation program could not retrieve version information from the host. With this update, the installation program does not attempt to retrieve version information from ESXi hosts that are in maintenance mode, allowing the installation to proceed. (link:https://issues.redhat.com/browse/OCPBUGS-31387[*OCPBUGS-31387*])
```

Author: :heavy_check_mark:

[id="ocp-4-15-18-unservable-future-versions-status-condition"]
===== Introducing the UnservableFutureVersions status condition for SHA1 routes

> * {product-title} {product-version} does not support SHA1 certificates on routes. Consequently, upgrades from 4.15 to {product-version} will cause routes with SHA1 certificates to be rejected.

Reviewer 2: ```suggestion
* {product-title} {product-version} does not support SHA1 certificates on routes. Consequently, upgrades from 4.15 to {product-version} causes routes with SHA1 certificates to be rejected.
```
